.github/workflows/ci-on-push.yml

```diff
@@ -0,0 +1,71 @@
+name: Continuous integration
+
+on: push
+
+jobs:
+  test_lint_deploy:
+    name: 'Test, lint and deploy'
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Get yarn cache directory path
+      id: yarn-cache-dir-path
+      run: echo "::set-output name=dir::$(yarn cache dir)"
+    - uses: actions/cache@v2
+      id: yarn-cache # use this to check for `cache-hit` (`steps.yarn-cache.outputs.cache-hit != 'true'`)
+      with:
+        path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+        key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-yarn-
+    - run: yarn --prefer-offline
+    - run: yarn run type-check && yarn run build
+    - run: yarn run lint
+    - run: yarn run check:custom-properties
+    - run: yarn run test --coverage && yarn run codecov
+    - run: node ./scripts/build-shrink-ray.js
```

Comment:
This runs synchronously, I experimented with matrices but that creates multiple containers.

Reply:
I think synchronous is fine. Each of these steps are pretty quick to run, we don't save much time by making them run in parallel

typecheck & build: 1m26s
lint: 1m53s
custom properties check: 8s
tests with coverage: 3m6s
shrink ray: 1m36s

That adds up to 8mins (with a cold cache, we could get some speedups by caching the build folder I think). If we did one thing per container (as travis does now) then total time is the time of the slowest item: 3mins. Which sounds good but meanwhile the a11y tests still take 10mins so we'd still be waiting for that even after all these run in parallel.

I don't think the extra complication and extra resources (1 action container vs 5) is worth the time "saved".


---

Question: Does a step item still run if the prior item failed? For instance currently we run linting then tests - if linting fails do the tests still run. I'm wondering if I commit something that fails both tests and linting am i alerted to both failures or just the linting?

Reply:
No they all stop if one fails. Same with matrices. The only way to get around that is separate containers from my understanding.

Reply:
The other option is making multiple `.yml` files and treating them as individual workflows.

Reply:
Looks like you can modify that behaviour: https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions#job-status-check-functions

```yaml
- if: ${{ !cancelled() }}
  run: yarn run type-check && yarn run build
- if: ${{ !cancelled() }}
  run: yarn run lint
- if: ${{ !cancelled() }}
  run: yarn run check:custom-properties
// etc
```